egs/musdb18/OpenUnmix/test_dataloader.py

```diff
@@ -0,0 +1,134 @@
+from asteroid.data import MUSDB18Dataset
+import argparse
+import torch
+from pathlib import Path
+import tqdm
+
+
+class Compose(object):
+    """Composes several augmentation transforms.
+    Args:
+        augmentations: list of augmentations to compose.
+    """
+
+    def __init__(self, transforms):
+        self.transforms = transforms
+
+    def __call__(self, audio):
+        for transform in self.transforms:
+            audio = transform(audio)
+        return audio
+
+
+def _augment_gain(audio, low=0.25, high=1.25):
+    """Applies a random gain to each source between `low` and `high`"""
+    gain = low + torch.rand(1) * (high - low)
+    return audio * gain
+
+
+def _augment_channelswap(audio):
+    """Randomly swap channels of stereo sources"""
+    if audio.shape[0] == 2 and torch.FloatTensor(1).uniform_() < 0.5:
+        return torch.flip(audio, [0])
+
+    return audio
+
+
+if __name__ == "__main__":
+    """dataset tests -> these parameters will go into recipes"""
+    parser = argparse.ArgumentParser(description='MUSDB18 dataset test')
+
+    parser.add_argument(
+        '--root', type=str, help='root path of dataset'
+    )
+
+    parser.add_argument('--seed', type=int, default=42)
+
+    parser.add_argument(
+        '--seq-dur', type=float, default=6.0,
+        help='Duration of <=0.0 will result in the full audio'
+    )
+
+    parser.add_argument(
+        '--samples-per-track', type=int, default=64,
+        help='draws a fixed number of samples per track'
+    )
+
+    parser.add_argument(
+        '--random-track-mix',
+        action='store_true', default=False,
+        help='Apply random track mixing augmentation'
+    )
+    parser.add_argument(
+        '--source-augmentations', type=str, nargs='+',
+        default=['gain', 'channelswap']
+    )
+    parser.add_argument('--batch-size', type=int, default=16)
+
+    args = parser.parse_args()
+
+    dataset_kwargs = {
+        'root': Path(args.root),
+    }
+
+    source_augmentations = Compose(
+        [globals()['_augment_' + aug] for aug in args.source_augmentations]
+    )
```

Comment:
I like that. We would probably opt for a `get` function for augmentations, as we have for norms, filterbanks etc..

Reply:
I would change this later when you've added the Compose class to asteroid core